### PR TITLE
Fix protocol parameters integer types

### PR DESCRIPTION
**Pull request by thibault-martinez**
Closes https://github.com/iotaledger/iota-sdk/issues/1072

### CODE REVIEW

File: sdk/src/types/block/protocol.rs [hunk 1]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
     /// time.
     pub(crate) max_committable_age: SlotIndex,
-    /// TODO
+    /// Epoch Nearing Threshold is used by the epoch orchestrator to detect the slot that should trigger a new
+    /// committee selection for the next and upcoming epoch.
     pub(crate) epoch_nearing_threshold: SlotIndex,
     /// Parameters used to calculate the Reference Mana Cost (RMC).
     pub(crate) congestion_control_parameters: CongestionControlParameters,
-    /// TODO
+    /// Version Signaling defines the parameters used by signaling protocol parameters upgrade.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// Defines the parameters used to signal a protocol parameters upgrade.
```

File: sdk/src/types/block/protocol.rs [hunk 2]
```diff
@@ -285,7 +287,6 @@ impl Default for CongestionControlParameters {
     }
 }
 
-// TODO docs
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs:288]: but no docs :c

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs:288]: Oh sorry, removed it because of the fields but forgot the main one, will add later

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs:288]: Done

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/types/block/protocol.rs [hunk 3]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
+    pub(crate) mana_decay_factors_exponent: u8,
     /// An integer approximation of the sum of decay over epochs.
     pub(crate) mana_decay_factor_epochs_sum: u32,
     /// The scaling of `mana_decay_factor_epochs_sum` expressed as an exponent of 2.
-    pub(crate) mana_decay_factor_epochs_sum_exponent: u32,
+    pub(crate) mana_decay_factor_epochs_sum_exponent: u8,
     /// The unbonding period in epochs before an account can stop staking.
     pub(crate) staking_unbonding_period: EpochIndex,
-    /// TODO
+    /// Liveness Threshold is used by tip-selection to determine the if a block is eligible by evaluating issuingTimes
```
Comment (kwek20) [inline sdk/src/types/block/protocol.rs]: To determine the if? and `issuingTimes`

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs]: will remove the `the`, can you comment on the TIP pls?

Comment (thibault-martinez) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// Used by tip-selection to determine if a block is eligible by evaluating issuing times.
```

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs]: Changed

File: sdk/src/types/block/protocol.rs [hunk 4]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
+    /// and commitments in its past-cone to Accepted Tangle Time and lastCommittedSlot respectively.
     pub(crate) liveness_threshold: SlotIndex,
     /// Minimum age relative to the accepted tangle time slot index that a slot can be committed.
     pub(crate) min_committable_age: SlotIndex,
     /// Maximum age for a slot commitment to be included in a block relative to the slot index of the block issuing
     /// time.
     pub(crate) max_committable_age: SlotIndex,
-    /// TODO
+    /// Epoch Nearing Threshold is used by the epoch orchestrator to detect the slot that should trigger a new
+    /// committee selection for the next and upcoming epoch.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// The slot index used by the epoch orchestrator to detect the slot that should trigger a new
    /// committee selection for the next and upcoming epoch.
```

File: sdk/src/types/block/protocol.rs [hunk 5]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
+    pub(crate) mana_decay_factors_exponent: u8,
     /// An integer approximation of the sum of decay over epochs.
     pub(crate) mana_decay_factor_epochs_sum: u32,
     /// The scaling of `mana_decay_factor_epochs_sum` expressed as an exponent of 2.
-    pub(crate) mana_decay_factor_epochs_sum_exponent: u32,
+    pub(crate) mana_decay_factor_epochs_sum_exponent: u8,
     /// The unbonding period in epochs before an account can stop staking.
     pub(crate) staking_unbonding_period: EpochIndex,
-    /// TODO
+    /// Used by tip-selection to determine if a block is eligible by evaluating issuing times.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// The slot index used by tip-selection to determine if a block is eligible by evaluating issuing times
```

File: sdk/src/types/block/protocol.rs [hunk 6]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
+    /// and commitments in its past-cone to Accepted Tangle Time and lastCommittedSlot respectively.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// and commitments in its past-cone to accepted tangle time and last committed slot respectively.
```

File: sdk/src/types/block/protocol.rs [hunk 7]
```diff
@@ -295,9 +297,13 @@ impl Default for CongestionControlParameters {
 #[packable(unpack_error = Error)]
 #[getset(get_copy = "pub")]
 pub struct VersionSignalingParameters {
-    window_size: u32,
-    window_target_ratio: u32,
-    activation_offset: u32,
+    /// The size of the window in epochs to find which version of protocol parameters was most signaled, from
+    /// currentEpoch - windowSize to currentEpoch.
```
Comment (kwek20) [inline sdk/src/types/block/protocol.rs]: param names here

Comment (thibault-martinez) [reply sdk/src/types/block/protocol.rs]: Changed

**Review by abdulmth — DISMISSED** [review]

File: sdk/src/types/block/protocol.rs [hunk 8]
```diff
@@ -45,39 +45,41 @@ pub struct ProtocolParameters {
...
     /// An integer approximation of the sum of decay over epochs.
     pub(crate) mana_decay_factor_epochs_sum: u32,
     /// The scaling of `mana_decay_factor_epochs_sum` expressed as an exponent of 2.
-    pub(crate) mana_decay_factor_epochs_sum_exponent: u32,
+    pub(crate) mana_decay_factor_epochs_sum_exponent: u8,
     /// The unbonding period in epochs before an account can stop staking.
     pub(crate) staking_unbonding_period: EpochIndex,
-    /// TODO
+    /// The slot index used by tip-selection to determine if a block is eligible by evaluating issuing times
+    /// and commitments in its past-cone to accepted tangle time and last committed slot respectively.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// and commitments in its past-cone against accepted tangle time and last committed slot respectively.
```

Sorry, re-reading this I have one more suggestion

File: sdk/src/types/block/protocol.rs [hunk 9]
```diff
@@ -295,9 +297,13 @@ impl Default for CongestionControlParameters {
 #[packable(unpack_error = Error)]
 #[getset(get_copy = "pub")]
 pub struct VersionSignalingParameters {
-    window_size: u32,
-    window_target_ratio: u32,
-    activation_offset: u32,
+    /// The size of the window in epochs to find which version of protocol parameters was most signaled, from
+    /// current_epoch - window_size to current_epoch.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// The size of the window in epochs that is used to find which version of protocol parameters was 
    /// most signaled, from `current_epoch - window_size` to `current_epoch`.
```

File: sdk/src/types/block/protocol.rs [hunk 10]
```diff
@@ -295,9 +297,13 @@ impl Default for CongestionControlParameters {
...
+    window_size: u8,
+    /// The target number of supporters for a version to win in a window_size.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// The number of supporters required for a version to win within a `window_size`.
```

File: sdk/src/types/block/protocol.rs [hunk 11]
```diff
@@ -295,9 +297,13 @@ impl Default for CongestionControlParameters {
...
+    window_target_ratio: u8,
+    /// The offset in epochs to activate the new version of protocol parameters.
```
Comment (DaughterOfMars) [inline sdk/src/types/block/protocol.rs]:
```suggestion
    /// The offset in epochs required to activate the new version of protocol parameters.
```

**Review by DaughterOfMars — DISMISSED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]
we did it!

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**